src/Microsoft.ML.Transforms/LearnerFeatureSelection.cs

```diff
@@ -32,9 +33,11 @@ public sealed class Arguments
             [Argument(ArgumentType.AtMostOnce, HelpText = "The number of slots to preserve", ShortName = "topk", SortOrder = 1)]
             public int? NumSlotsToKeep;
 
-            [Argument(ArgumentType.Multiple, HelpText = "Filter", ShortName = "f", SortOrder = 1)]
-            public SubComponent<ITrainer<IPredictorWithFeatureWeights<Single>>, SignatureFeatureScorerTrainer> Filter =
-                new SubComponent<ITrainer<IPredictorWithFeatureWeights<Single>>, SignatureFeatureScorerTrainer>("SDCA");
+            [Argument(ArgumentType.Multiple, HelpText = "Filter", ShortName = "f", SortOrder = 1, SignatureType = typeof(SignatureFeatureScorerTrainer))]
+            public IComponentFactory<ITrainer<IPredictorWithFeatureWeights<Single>>> Filter =
+                ComponentFactoryUtils.CreateFromFunction(env =>
+                    // ML.Transforms doesn't have a direct reference to ML.StandardLearners, so use ComponentCatalog to create the Filter
```

Comment:
>// ML.Transforms doesn't have a direct reference to ML.StandardLearners, [](start = 20, length = 72)

since it's a learner feature selection, maybe make sense to move it to learners project?

Reply:
I've opened [https://github.com/dotnet/machinelearning/issues/726](https://github.com/dotnet/machinelearning/issues/726) for this discussion.  We can decide what to do in that issue.  For now, keeping it to use DI (like it was before) is a decent approach.

---
In reply to: [211692955](https://github.com/dotnet/machinelearning/pull/700#discussion_r211692955) [](ancestors = 211692955)